tests/filter-tests.js

```diff
@@ -521,4 +521,75 @@ describe('broccoli-asset-rev', function () {
+    const subject = new AssetRewrite(input.path(), {
+      assetMap: {
+        '/assets/img/small.png': '/assets/img/other-small.png',
+        '/assets/img/medium.png': '/assets/img/other-medium.png',
+        '/assets/img/big.png': '/assets/img/other-big.png',
+      },
+    });
+    const output = createBuilder(subject);
+    input.write({
+      'srcset.html': `<source srcset="/assets/img/small.png 800w, /assets/img/medium.png 1600w" sizes="(max-width: 800px) 800px, 1600px" type="image/png">`,
+    });
+
+    await output.build();
+    expect(output.changes()).to.deep.equal({
+      'srcset.html': 'create',
+    });
+    expect(output.read()).to.deep.equal({
+      'srcset.html': `<source srcset="/assets/img/other-small.png 800w, /assets/img/other-medium.png 1600w" sizes="(max-width: 800px) 800px, 1600px" type="image/png">`,
+    });
+  });
+
+  it('replaces assets in srcset attributes with prepended path', async function () {
+    const input = await createTempDir();
+    const subject = new AssetRewrite(input.path(), {
+      assetMap: {
+        '/assets/img/small.png': '/assets/img/other-small.png',
+        '/assets/img/medium.png': '/assets/img/other-medium.png',
+        '/assets/img/big.png': '/assets/img/other-big.png',
+      },
+      prepend: 'my-path/',
+    });
+    const output = createBuilder(subject);
+    input.write({
+      'srcset.html': `<source srcset="/assets/img/small.png 800w, /assets/img/medium.png 1600w" sizes="(max-width: 800px) 800px, 1600px" type="image/png">`,
+    });
+
+    await output.build();
+    expect(output.changes()).to.deep.equal({
+      'srcset.html': 'create',
+    });
+    expect(output.read()).to.deep.equal({
+      'srcset.html': `<source srcset="my-path/assets/img/other-small.png 800w, my-path/assets/img/other-medium.png 1600w" sizes="(max-width: 800px) 800px, 1600px" type="image/png">`,
+    });
+  });
+
+  it('replaces assets in srcset attributes with prepended domain', async function () {
+    const input = await createTempDir();
+    const subject = new AssetRewrite(input.path(), {
+      assetMap: {
+        '/assets/img/small.png': '/assets/img/other-small.png',
+        '/assets/img/medium.png': '/assets/img/other-medium.png',
+        '/assets/img/big.png': '/assets/img/other-big.png',
+      },
+      prepend: 'https://subdomain.cloudfront.net/',
+    });
+    const output = createBuilder(subject);
+    input.write({
+      'srcset.html': `<source srcset="/assets/img/small.png 800w, /assets/img/medium.png 1600w" sizes="(max-width: 800px) 800px, 1600px" type="image/png">`,
+    });
+
+    await output.build();
+    expect(output.changes()).to.deep.equal({
+      'srcset.html': 'create',
+    });
+    expect(output.read()).to.deep.equal({
+      'srcset.html': `<source srcset="https://subdomain.cloudfront.net/assets/img/other-small.png 800w, https://subdomain.cloudfront.net/assets/img/other-medium.png 1600w" sizes="(max-width: 800px) 800px, 1600px" type="image/png">`,
+    });
+  });
```

Comment:
@st-h how about adding a test for the line break issue?

Reply:
Yes. Good call. Just haven't had any time to look into this further.